csv-generator/csv_generator/article_pub_dates.py

```diff
@@ -0,0 +1,78 @@
+import argparse
+import csv
+import logging
+import os
+from tempfile import TemporaryDirectory
+
+import requests
+from tqdm import tqdm
+
+from consumers.utils import timestamp_to_epoch
+
+
+LOGGER = logging.getLogger(__name__)
+
+CSV_TEMP_FILE = 'temp.csv'
+CSV_OUTPUT_FILE = 'published_article_index.csv'
+DEFAULT_OUTPUT_DIR = '../output'
+HEADERS = ['msid', 'poa_published_date', 'vor_published_date']
+URL = 'https://observer.elifesciences.org/report/published-research-article-index.csv'
+
+
+def _get_article_pub_index() -> str:
+    response = requests.get(URL)
+    if response.status_code == 200:
```

Comment:
should it maybe throw an exception if it fails? I usually use `response.raise_for_status` but since you are using a temporary file anyway `urlretrieve` might be more convenient?

Reply:
Will add in `response.raise_for_status`.